cli/src/main.rs

```diff
@@ -44,10 +39,30 @@ struct Args {
     dev: bool,
 }
 
+// Infer the output format based on the file extension of the output format
+fn infer_output_format(output: &Path) -> Option<OutputFormat> {
+    output
+        .extension()
+        .map(|ext| match ext.to_str() {
+            Some("tex") => Some(OutputFormat::new("latex")),
+            Some("html") => Some(OutputFormat::new("html")),
+            _ => None,
+        })
+        .flatten()
+}
+
 fn compile_file(args: &Args) -> Result<Ast, CliError> {
     let source = fs::read_to_string(&args.input)?;
     let mut ctx = Context::default();
-    let output = eval(&source, &mut ctx, &OutputFormat::new(&args.format))?;
+
+    let Some(format) = args.format
+        .as_ref()
+        .map(|s| OutputFormat::new(&s))
```

Comment:
Hmm, you do `as_ref` and then take a ref, seems to me that you can replace `&s` to `s`, and then u are fine with just giving the function handle directly

Reply:
Nope, sadly it doesn't work since `OutputFormat::new` expects a `&str` and I have `&String`.

Reply:
It is the same thing with automatic coersion, and `.map(|s| OutputFormat::new(s))` compiles on my machine